Initial Commit of Request Addon

Selindrile opened the pull request:
Request allows other players to send you commands using a whitelist and
blacklist for security with optional partial disables.

Selindrile committed:
Initial Commit of Request Addon

Request allows other players to send you commands using a whitelist and
blacklist for security with optional partial disables.

Selindrile committed:
Fixed Variables, Errors, Comments.

Made nick a local variable.
Made it so if a variable is nil, sets it to none before processing to
prevent errors.

Selindrile committed:
Added Forbidden.

Further fine tuned command capture and added the option to forbid
specific commands.

Selindrile committed:
Added default settings.

Added default forbidden commands.

Selindrile committed:
Fixed Typo

Selindrile committed:
Added P to forbidden.

Selindrile committed:
Added Unloadall to forbidden.

Selindrile committed:
Merge remote-tracking branch 'upstream/4.1-dev' into 4.1-dev

Selindrile commented:
Added a forbidden command list as a safety and personal preference feature.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+--[[
+Copyright (c) 2015, Selindrile
```
Change the `(c)` to `©`.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of autoinvite nor the
```
This is not AutoInvite :)

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+require('luau')
+
+_addon.name = 'Request'
+_addon.author = 'Selindrile, Supported by Booshack: Montaeg (LUA), Hrothgar & Terezka (Beta Testers)'
```
Make this shorter, just your name is fine. Otherwise it will go off the screen of other players.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+_addon.language = 'english'
+
+defaults = T{}
+defaults.mode = 'whitelist'
```
We conventionally use _PascalCase_ for settings.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+    blist        = 'blacklist',
+    black        = 'blacklist',
+    blacklist    = 'blacklist',
+	nick		 = 'nicknames',
```
Please use 4 space indentation everywhere in the code.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+    local doubles = names * settings[mode]
+    if not doubles:empty() then
+        if aliases[mode] == 'nicknames' then
+            notice('nickname':plural(doubles)..' '..doubles:format()..' already on nickname list.')
```
The first `nickname` here should be `User`.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+        if aliases[mode] == 'nicknames' then
+            notice('nickname':plural(doubles)..' '..doubles:format()..' already on nickname list.')
+        elseif aliases[mode] == 'forbidden' then
+			notice('forbidden':plural(doubles)..' '..doubles:format()..' already on forbidden list.')
```
The first `forbidden` here should be `User`.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+    local dummy = names - settings[mode]
+    if not dummy:empty() then
+        if aliases[mode] == 'nicknames' then
+            notice('nickname':plural(dummy)..' '..dummy:format()..' not found on nickname list.')
```
The first `nickname` here should be `User`.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+        if aliases[mode] == 'nicknames' then
+            notice('nickname':plural(dummy)..' '..dummy:format()..' not found on nickname list.')
+        elseif aliases[mode] == 'forbidden' then
+			notice('forbidden':plural(dummy)..' '..dummy:format()..' not found on forbidden list.')
```
The first `forbidden` here should be `User`.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+    end
+
+    config.save(settings)
+end)
```
Newline at the end.

z16 commented on addons/Request/data/settings.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.1" ?>
```
Remove this file from the commit. It will (and should) be automatically generated.

Selindrile replied:
I want it to already have those blacklisted commands that most would want to blacklist already in the settings before they add their settings to it, they can remove them if they wish, but I think having it there is better than not.

Selindrile committed:
Update Request.lua

Selindrile committed:
Update Request.lua

Selindrile committed:
Update settings.xml

Selindrile committed:
Update Request.lua

Selindrile committed:
Update Request.lua

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+defaults.whitelist = S{}
+defaults.blacklist = S{}
+defaults.nicknames = S{}
+defaults.forbidden = S{}
```
Add some commands here if you want them to be forbidden by default instead of creating a `settings.xml` file.

Selindrile replied:
Ah, I thought doing that would make them forbidden on every load even if the user had removed them, thanks.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+defaults.blacklist = S{}
+defaults.nicknames = S{}
+defaults.forbidden = S{}
+defaults.partylock = 'On'
```
These should be `true` or `false`.

Selindrile replied:
I'd rather them be on or off so that they're displayed as such when the user views them with //request but I suppose I could write an if statement that just displays on or off if it's true or false if you'd rather that.

z16 replied:
You can make a function like this:

``` lua
display = function(setting)
    if class(setting) == 'Set' then
        return setting:empty() and '(empty)' or setting:format('csv')
    elseif class(setting) == 'boolean' then
        return setting and 'on' or 'off'
    end

    return tostring(setting)
end
```

You can also adjust your status display function:

``` lua
    elseif command == 'status' then
        for key in pairs(defaults) do
            log('%s: %s':format(key, settings[key]))
        end
    end
```

That way you can add new settings at any time and it'll display them correctly without you manually having to add them to the status list.

z16 commented on addons/Request/Request.lua:
```diff
@@ -0,0 +1,305 @@
+defaults.forbidden = S{}
+defaults.partylock = 'On'
+defaults.exactlock = 'On'
+defaults.requestlock = 'Off'
```
You should still use _PascalCase_ for settings.

Selindrile replied:
I thought you meant for what was contained within partylock/exactlock/requestlock, so that it would show up capitalized when a user viewed the settings with //request, but I see you asked me to use true or false there. Did you mean for me to pascalcase: defaults.PartyLock, defaults.ExactLock, defaults.RequestLock like so?

z16 replied:
Correct.

z16 commented on addons/Request/data/settings.xml:
```diff
@@ -0,0 +1,12 @@
+<settings>
```
This file still needs to go. See the above notes on the default values.

Selindrile committed:
Edits.

I pascalcased the three lock settings, but I couldn't get the whitelist/blacklist/mode to capitalize without breaking things even making sure to capitalize all the mentions of them I could find and seemed to be causing errors that were very unexpected to me, hoping you just meant the boolean settings anyway.

Selindrile committed:
Delete settings.xml

alphaONE2 commented:
Is there anything blocking this? It's been open for a while now with no activity.

_P.S. I find this addon somewhat questionable, but I'm not sure I'm willing to go as far as refusing the submission._

z16 commented:
I don't!

Selindrile commented:
Honestly I gave up on standardizing it as per Arcon's request, have been keeping my personal version updated but, while it sounds stupid, something as easy as pascalcasing the settings seems to break it every time, I gave up.

z16 commented:
Changing the settings doesn't work right after people start using it once (not unless you nuke them entirely anyway). That's something that needs to be done before the addon's initial commit.

Selindrile commented:
Yeah, I forgot to nuke the settings.xml, builds from the lua now.
